src/query/functions/src/scalars/variant.rs

```diff
@@ -1057,3 +1081,25 @@ fn json_object_fn(
         None => Value::Scalar(Scalar::Variant(builder.build_scalar())),
     }
 }
+
+fn prepare_args_columns(
+    args: &[ValueRef<AnyType>],
+    ctx: &EvalContext,
+) -> (Vec<Column>, Option<usize>) {
+    let len = args.iter().find_map(|arg| match arg {
+        ValueRef::Column(col) => Some(col.len()),
+        _ => None,
+    });
+    let mut columns = Vec::with_capacity(args.len());
+    for (i, arg) in args.iter().enumerate() {
+        let column = match arg {
+            ValueRef::Column(column) => column.clone(),
+            ValueRef::Scalar(s) => {
+                let column_builder = ColumnBuilder::repeat(s, 1, &ctx.generics[i]);
```

Comment:
The `num_rows` should be the length of the column. Sorry, this is an existing bug.

Reply:
Did you mean that it should be? 
`
ColumnBuilder::repeat(s, len.unwrap_or(1), &ctx.generics[i])
`

Reply:
yes